From af139ec373c203b22891e0a82614b22f01df7266 Mon Sep 17 00:00:00 2001
From: "karthikgk@maxlinear.com" <karthikgk@lvdocker7.lvsw.maxlinear.com>
Date: Fri, 29 Aug 2025 01:50:55 -0700
Subject: [PATCH] WLANRTSYS-92335 : WLAN driver r1233-3bf57a fail to build
 iwlwav-hostap-ng

Issue : This error occurs because the compiler sees two different definitions of struct wpa_bss
 In the header file (mxl_supplicant.h): The compiler doesn't know what struct wpa_bss is, so it creates an incomplete/local type definition
 In the source file (mxl_supplicant.c): The compiler has the complete definition of struct wpa_bss from included headers

Fix : Including bss.h header file before driver_i.h will resolve this
issue.

signed-off-by : karthikgk <karthikgk@maxlinear.com>
---
 wpa_supplicant/mxl_supplicant.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wpa_supplicant/mxl_supplicant.c b/wpa_supplicant/mxl_supplicant.c
index 55336d370..d36881ad3 100644
--- a/wpa_supplicant/mxl_supplicant.c
+++ b/wpa_supplicant/mxl_supplicant.c
@@ -23,8 +23,8 @@
 #include "config.h"
 #include "ap/mxl_hostapd.h"
 #include "mxl_supplicant_config.h"
-#include "driver_i.h"
 #include "bss.h"
+#include "driver_i.h"
 #include "wpa_supplicant_i.h"
 
 #ifdef CONFIG_VENDOR_MXL_PROPRIETARY
-- 
2.43.0

